src/realm/array.cpp

```diff
@@ -427,8 +533,14 @@ size_t Array::blob_size() const noexcept
 
 void Array::insert(size_t ndx, int_fast64_t value)
 {
-    REALM_ASSERT_DEBUG(ndx <= m_size);
+    decode_array();
```

Comment:
This breaks a lot of stuff, but eventually wouldn't we want to do this in the end? Probably before to decode, we need to add a check in order to evaluate if the array is encoded and if adding the new element will still make sense.